vm/vm.go

```diff
@@ -324,23 +331,23 @@ func (v *vm) Execute(txns []core.Transaction, declaredClasses []core.Class, paid
 
 	if context.err != "" {
 		if context.errTxnIndex >= 0 {
-			return nil, nil, nil, TransactionExecutionError{
+			return nil, nil, nil, context.executionSteps, TransactionExecutionError{
 				Index: uint64(context.errTxnIndex),
 				Cause: errors.New(context.err),
 			}
 		}
-		return nil, nil, nil, errors.New(context.err)
+		return nil, nil, nil, context.executionSteps, errors.New(context.err)
 	}
 
 	traces := make([]TransactionTrace, len(context.traces))
 	for index, traceJSON := range context.traces {
 		if err := json.Unmarshal(traceJSON, &traces[index]); err != nil {
-			return nil, nil, nil, fmt.Errorf("unmarshal trace: %v", err)
+			return nil, nil, nil, context.executionSteps, fmt.Errorf("unmarshal trace: %v", err)
```

Comment:
in case of errors we return zero values (i.e. `0`)

Reply:
even if the vm ran for some time?

Reply:
even in cases where the errors came as a result of the transactions
not the system?

Reply:
> even in cases where the errors came as a result of the transactions
not the system?

yes, in general we should use this approach